mods/fully-customizable-winver.wh.cpp

```diff
@@ -0,0 +1,1260 @@
+    if (Wh_GetIntSetting(L"windowSize.enabled") && Wh_GetIntSetting(L"windowSize.expandControls")) {
+        Wh_Log(L"EXPAND CONTROLS");
+        RECT windowRect;
+        GetWindowRect(hWnd, &windowRect);
+        INT diffX = (windowRect.right - windowRect.left) - (originalWindowRect.right - originalWindowRect.left);
+        INT diffY = (windowRect.bottom - windowRect.top) - (originalWindowRect.bottom - originalWindowRect.top);
+        Wh_Log(L"Difference between old and new size: x -> %d, y -> %d", diffX, diffY);
+        for (auto& control : g_defaultWindowAttributes) {
+            if (control.first == L"LeftIcon") {
+                continue;
+            }
+            HWND controlHWND = control.second.hWnd;
+            if (IsWindow(controlHWND)) {
+                RECT rect;
+                GetWindowRect(controlHWND, &rect);
+                MapWindowPoints(nullptr, hWnd, reinterpret_cast<POINT*>(&rect), 2);
+                if (control.first == L"OkButton") {
+                    rect.left += diffX;
+                    if (Wh_GetIntSetting(L"windowSize.exceptionalOkButton")) {
+                        rect.bottom += diffY;
+                        rect.top += diffY;
+                    }
+                }
+                MoveWindow(controlHWND, rect.left, rect.top, rect.right - rect.left + diffX, rect.bottom - rect.top, TRUE);
+            }
+        }
+    }
+}
+
+void Update(HWND hWnd) {
+    Wh_Log(L"UPDATING CUSTOM ICON");
+    SetCustomIcon(hWnd);
+    Wh_Log(L"UPDATING POSITION");
+    UpdatePosition(hWnd);
+    Wh_Log(L"UPDATING CONTROLS");
+    UpdateControls(hWnd);
+    Wh_Log(L"UPDATING TITLE");
+    UpdateTitle(hWnd);
+    Wh_Log(L"UPDATING WINDOW COLORS");
+    UpdateWindowColors(hWnd);
+    Wh_Log(L"UPDATING LOGO");
+    UpdateLogo(hWnd);
+    Wh_Log(L"DONE");
+}
+
+void ShowErrorsIfAny(HWND hWnd) {
+    for (auto& message : g_errorMessages) {
+        MessageBoxW(hWnd, message.c_str(), MOD_TITLE, MB_OK | MB_ICONERROR);
+    }
+    g_errorMessages.clear();
+}
+
+HBITMAP CapturePixelsToBitmap(const RECT& rect)
+{
+    RECT clientRect{};
+    GetClientRect(g_mainWindow, &clientRect);
+    const INT W = clientRect.right;
+    const INT H = clientRect.bottom;
+
+    HDC hdc = GetDC(g_mainWindow);
+    if (!hdc) {
+        return nullptr;
+    }
+
+    BITMAPINFO bmi{};
+    bmi.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+    bmi.bmiHeader.biWidth = W;
+    bmi.bmiHeader.biHeight = -H;
+    bmi.bmiHeader.biPlanes = 1;
+    bmi.bmiHeader.biBitCount = 32;
+    bmi.bmiHeader.biCompression = BI_RGB;
+
+    void* fullBits = nullptr;
+    HBITMAP fullBmp = CreateDIBSection(hdc, &bmi, DIB_RGB_COLORS, &fullBits, nullptr, 0);
+    if (!fullBmp) {
+        ReleaseDC(g_mainWindow, hdc);
+        return nullptr;
+    }
+
+    HDC fullDC = CreateCompatibleDC(hdc);
+    HGDIOBJ oldFull = SelectObject(fullDC, fullBmp);
+
+    BOOL ok = FALSE;
+#if defined(PW_RENDERFULLCONTENT)
```

Comment:
Why is this necessary? If it means that different code will be generated for different Windhawk versions, then just define `PW_RENDERFULLCONTENT` if it's not defined, like:
```cpp
#ifndef PW_RENDERFULLCONTENT
#define PW_RENDERFULLCONTENT 0x00000002
#endif
```